File: NATSInternal.Web/Models/Shared/ProductCategoryBasicModel.cs
using NATSInternal.Application.UseCases.Shared;

namespace NATSInternal.Web.Models;

public class ProductCategoryBasicModel
{
    #region Constructors
    public ProductCategoryBasicModel(ProductCategoryBasicResponseDto responseDto)
    {
        Id = responseDto.Id;
        Name = responseDto.Name;
    }
    #endregion

    #region Properties
    public Guid Id { get; set; }
    public string Name { get; set; }
    #endregion
}

File: NATSInternal.Frontend/NATSInternal.Vue/src/models/shared/productCategoryBasicModel.ts
declare global {
  type ProductCategoryBasicModel = {
    id: string;
    name: string;
  };
}

function createProductCategoryBasicModel(responseDto: ProductCategoryBasicResponseDto): ProductCategoryBasicModel {
  return {
    id: responseDto.id,
    name: responseDto.name
  };
}

export { createProductCategoryBasicModel };